pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -180,47 +191,57 @@ public synchronized void refreshSegment(String segment) {
       }
       Set<String> selectedSegments = new HashSet<>();
       for (String segment : segments) {
-        PartitionInfo partitionInfo = _partitionInfoMap.get(segment);
-        if (partitionInfo == null || partitionInfo == INVALID_PARTITION_INFO || isPartitionMatch(filterQueryTree,
-            partitionInfo)) {
+        Map<String, PartitionInfo> columnPartitionInfo = _segmentColumnPartitionInfoMap.get(segment);
+        if (columnPartitionInfo == null || columnPartitionInfo == INVALID_COLUMN_PARTITION_INFO || isPartitionMatch(
+            filterQueryTree, columnPartitionInfo)) {
           selectedSegments.add(segment);
         }
       }
       return selectedSegments;
     }
   }
 
-  private boolean isPartitionMatch(Expression filterExpression, PartitionInfo partitionInfo) {
+  @VisibleForTesting
+  public Set<String> getPartitionColumns() {
+    return _partitionColumns;
+  }
+
+  private boolean isPartitionMatch(Expression filterExpression, Map<String, PartitionInfo> columnPartitionInfo) {
     Function function = filterExpression.getFunctionCall();
     FilterKind filterKind = FilterKind.valueOf(function.getOperator());
     List<Expression> operands = function.getOperands();
     switch (filterKind) {
       case AND:
         for (Expression child : operands) {
-          if (!isPartitionMatch(child, partitionInfo)) {
+          if (!isPartitionMatch(child, columnPartitionInfo)) {
             return false;
           }
         }
         return true;
       case OR:
         for (Expression child : operands) {
-          if (isPartitionMatch(child, partitionInfo)) {
+          if (isPartitionMatch(child, columnPartitionInfo)) {
             return true;
           }
         }
         return false;
       case EQUALS: {
         Identifier identifier = operands.get(0).getIdentifier();
-        if (identifier != null && identifier.getName().equals(_partitionColumn)) {
-          return partitionInfo._partitions.contains(
-              partitionInfo._partitionFunction.getPartition(operands.get(1).getLiteral().getFieldValue().toString()));
+        if (identifier != null && _partitionColumns.contains(identifier.getName())) {
```

Comment:
Save one extra map lookup
```suggestion
        if (identifier != null) {
```